src/include/daos_fs.h

```diff
@@ -902,6 +902,25 @@ dfs_access(dfs_t *dfs, dfs_obj_t *parent, const char *name, int mask);
 int
 dfs_chmod(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode);
 
+/**
+ * Change owner and group. Since uid and gid are not enforced
+ * at the DFS level, we do not also enforce the process privileges to be able to change the uid and
+ * gid. Any process with write access to the DFS container can make changes to the uid and gid using
+ * this function.
+ *
+ * \param[in]	dfs	Pointer to the mounted file system.
+ * \param[in]	parent	Opened parent directory object. If NULL, use root obj.
+ * \param[in]	name	Link name of the object. Can be NULL if parent is root,
+ *			which means operation will be on root object.
+ * \param[in]	uid	change owner of file (-1 to leave unchanged).
+ * \param[in]	gid	change group of file (-1 to leave unchanged).
+ * \param[in]	flags	if 0, symlinks are dereferenced. Pass O_NOFOLLOW to not dereference.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_chown(dfs_t *dfs, dfs_obj_t *parent, const char *name, uid_t uid, gid_t gid, int flags);
```

Comment:
From a dfuse and TOCTOU perspective I'd much rather tha dfs_ochown() function where you had to open something before you chown it.

It's easy to write a dfs_chown() as a wrapper around a dfs_ochown(), but it's impossible to do it the other way round safely.

Reply:
i guess we could add support for that in dfs_osettatr?
let me add that/

Reply:
Yes, that would be even better actually, as that's what dfuse already uses and then it can set uid/gid atomically with any other options.